### ISSUE TITLE
How to filter suggestions?

### ISSUE THREAD

**Issue by deltakosh** [microsoft/monaco-editor#17]
I would like to filter suggestions to remove properties and functions starting with _.

Is it possible?

**Comment by alexandrudima** [comment]
@jrieken @joaomoreno Do you know how/if this is feasible to add to the API ?

**Comment by jrieken** [comment]
That's not possible unless you tackle the provider of the suggestions, say TypeScript language service etc

**Comment by deltakosh** [comment]
It was possible in the preview versions

**Comment by deltakosh** [comment]
up?

**Comment by jrieken** [comment]
@deltakosh It is still not possible unless you use your own provider. Do you need help to implement/add your [own provider](https://github.com/Microsoft/vscode/blob/master/src/vs/monaco.d.ts#L3751)?

**Comment by deltakosh** [comment]
Any help would be greatly appreciated:) I just need the regular provider without _ properties and functions

**Comment by jrieken** [comment]
> I just need the regular provider without _ properties and functions

Do I assume correctly that you have added a `some_lib.d.ts` file that defines such symbols? Iff so why is it like that?

**Comment by deltakosh** [comment]
We added a custom d.ts where _ define private properties and functions

**Comment by jrieken** [comment]
> where _ define private properties and functions

So, why adding them and then filtering them again?

**Comment by deltakosh** [comment]
Do you really suggest that I manually remove all _propertiesfrom my d.ts?
Here is the file: https://github.com/BabylonJS/Babylon.js/blob/master/dist/babylon.2.4.d.ts

**Comment by jrieken** [comment]
No, my question is why does the `d.ts` contains symbols with `_` if those aren't desired. Do you control that API? Is it just a file you need to process? Is the API two fold and public layer and a private aka `_` layer?

You are failing to motivate why we should enable to filter suggestions globally in the editor while it seems to be you that's controlling those suggestions anyways?

**Comment by deltakosh** [comment]
I'm controlling this API but I know no straightforward way of removing _ layer.

But I'm certainly sure we are not the only API relying on this pattern.

So for all of these API we need to develop a tool to "clean" the definition file

**Comment by deltakosh** [comment]
Note: Some _ must be flagged as public as there is no internal in TS

**Comment by deltakosh** [comment]
Any thoughts?

**Comment by deltakosh** [comment]
I guess I failed to explain my issue. Se please let me try again:

- we are generating our Babylon.d.ts from our typescript source files
- private properties are by convention preceded by a _
- unfortunately as there is no internal keyword in typescript, we have to flag some _ properties as public 

Hopefully this is clearer now:)

**Comment by deltakosh** [comment]
up?

**Comment by deltakosh** [comment]
up?

**Comment by meulta** [comment]
Do you have news on this?

**Comment by davrous** [comment]
I'm also interested in this feature! Any news on that?

**Comment by julien-moreau** [comment]
Right ! Do you have any news about that ?

**Comment by sebavan** [comment]
I have the same issue, any ETA ?

**Comment by kpau** [comment]
> Note: Some _ must be flagged as public as there is no internal in TS

TypeScript compiler supports stripInternal option. It should work in your case.
http://www.typescriptlang.org/docs/handbook/compiler-options.html

**Comment by deltakosh** [comment]
The problem is that babylon.js library is pretty big now and adding /** @internal */ JSDoc annotation to all _ properties and methods is not an option

**Comment by deltakosh** [comment]
Apparently the 0.18 is now displaying properties flagged as private. Is that expected?

**Comment by atifsyedali** [comment]
In my case I just want to remove the `undefined` option from the list, without disabling variables. Don't want to implement a completion provider just to filter out one item from the list `undefined`.

**Comment by alexdima** [comment]
[This](https://github.com/microsoft/monaco-typescript/pull/65) might be helpful specifically for TypeScript.
